pkg/controller/controlplane/valuesprovider_test.go

```diff
@@ -269,13 +308,95 @@ var _ = Describe("ValuesProvider", func() {
+				"managedDefaultClass":  true,
+			}))
+		})
+
+		It("should return correct storage class chart values (k8s < 1.18) and default is set to true", func() {
+			cp.Spec.DefaultSpec.ProviderConfig.Raw = encode(&apisawsv1alpha1.ControlPlaneConfig{
+				Storage: &apisawsv1alpha1.Storage{
+					ManagedDefaultClass: pointer.BoolPtr(true),
+				},
+			})
+
+			values, err := vp.GetStorageClassesChartValues(ctx, cp, clusterK8sLessThan118)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(values).To(Equal(map[string]interface{}{
+				"useLegacyProvisioner": true,
+				"managedDefaultClass":  true,
+			}))
+		})
+
+		It("should return correct storage class chart values (k8s < 1.18) and default is set to false", func() {
+			cp.Spec.DefaultSpec.ProviderConfig.Raw = encode(&apisawsv1alpha1.ControlPlaneConfig{
+				Storage: &apisawsv1alpha1.Storage{
+					ManagedDefaultClass: pointer.BoolPtr(false),
+				},
+			})
+
+			values, err := vp.GetStorageClassesChartValues(ctx, cp, clusterK8sLessThan118)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(values).To(Equal(map[string]interface{}{
+				"useLegacyProvisioner": true,
+				"managedDefaultClass":  false,
+			}))
 		})
 
 		It("should return correct storage class chart values (k8s >= 1.18)", func() {
 			values, err := vp.GetStorageClassesChartValues(ctx, cp, clusterK8sAtLeast118)
 			Expect(err).NotTo(HaveOccurred())
-			Expect(values).To(Equal(map[string]interface{}{"useLegacyProvisioner": false}))
+			Expect(values).To(Equal(map[string]interface{}{
+				"useLegacyProvisioner": false,
+				"managedDefaultClass":  true,
+			}))
+		})
+
+		It("should return correct storage class chart values (k8s >= 1.18) and default is set to true", func() {
+			cp.Spec.DefaultSpec.ProviderConfig.Raw = encode(&apisawsv1alpha1.ControlPlaneConfig{
+				Storage: &apisawsv1alpha1.Storage{
+					ManagedDefaultClass: pointer.BoolPtr(true),
+				},
+			})
+
+			values, err := vp.GetStorageClassesChartValues(ctx, cp, clusterK8sAtLeast118)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(values).To(Equal(map[string]interface{}{
+				"useLegacyProvisioner": false,
+				"managedDefaultClass":  true,
+			}))
+		})
+
+		It("should return correct storage class chart values (k8s >= 1.18) and default is set to false", func() {
+			cp.Spec.DefaultSpec.ProviderConfig.Raw = encode(&apisawsv1alpha1.ControlPlaneConfig{
+				Storage: &apisawsv1alpha1.Storage{
+					ManagedDefaultClass: pointer.BoolPtr(false),
+				},
+			})
+
+			values, err := vp.GetStorageClassesChartValues(ctx, cp, clusterK8sAtLeast118)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(values).To(Equal(map[string]interface{}{
+				"useLegacyProvisioner": false,
+				"managedDefaultClass":  false,
+			}))
+		})
+
+		It("should have managedDefaultClass to true when using internal resource", func() {
+			internal := `{"kind":"ControlPlaneConfig","apiVersion":"aws.provider.extensions.gardener.cloud/__internal"}`
+
+			cp.Spec.DefaultSpec.ProviderConfig.Raw = []byte(internal)
```

Comment:
Can't this be written also as

```suggestion
			cp.Spec.DefaultSpec.ProviderConfig.Raw = encode(&apisaws.ControlPlaneConfig{})
```

Reply:
Encoder chooses external version over internal for the encoding. This won't work.